src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
+        match self.get_tag(collection_name).await {
+            Ok(hash) => Ok(hash),
+            Err(e) => {
+                println!("Error retrieving tag for collection: {:?}", e);
+                Err(anyhow!("Collection {} not found", collection_name))
+            }
+        }
+    }
+    pub async fn store_tag(&self, collection_name: &str, collection_hash: &Hash) -> Result<()> {
+        println!("Storing tag for collection: {} with hash: {:?}", collection_name, collection_hash);
+        
+        // Store the tag
+        self.store.set_tag(
+            collection_name.to_string().into(), 
+            Some(HashAndFormat::new(*collection_hash, BlobFormat::Raw))
+        ).await?;
+    
+        // Introducing a short delay to ensure that tag is persisted
+        tokio::time::sleep(Duration::from_millis(100)).await;
+    
+        // Verify that the tag was actually stored
+        match self.get_tag(collection_name).await {
+            Ok(stored_hash) => {
+                if stored_hash == *collection_hash {
+                    println!("Tag successfully stored for collection: {}", collection_name);
+                } else {
+                    return Err(anyhow!("Mismatch in stored tag for collection: {}", collection_name));
+                }
+            },
+            Err(e) => {
+                return Err(anyhow!("Failed to retrieve tag after storing for collection: {}. Error: {:?}", collection_name, e));
+            }
+        }
+        
+        Ok(())
+    }
+    
+    
+    
+    
+    pub async fn get_tag(&self, collection_name: &str) -> Result<Hash> {
+        println!("Retrieving tag for collection: {}", collection_name);
+        let mut tags = self.store.tags().await?;
+    
+           
+        while let Some(tag_result) = tags.next() {
+            match tag_result {
+                Ok((tag, hash_and_format)) => {
+                    // Logging for better insight
+                    println!("Checking tag: {:?}", tag);
+
+                    // Extract the bytes and convert them to a string
+                    let tag_bytes = tag.borrow();
+                    let tag_str = std::str::from_utf8(tag_bytes).unwrap_or("");
+
+                    // Compare the extracted tag string with collection_name
+                    if tag_str == collection_name {
+                        println!("Found matching tag for collection: {}", collection_name);
+                        return Ok(hash_and_format.hash);
+                    } else {
+                        println!("Tag {} does not match {}", tag_str, collection_name);
+                    }
+                }
+                Err(e) => {
+                    println!("Error reading tags: {:?}", e);
+                    return Err(anyhow!("Error reading tags: {:?}", e));
+                }
+            }
         }
+    
+        Err(anyhow!("Tag not found for collection: {}", collection_name))
+    }
+    
+    pub async fn list_tags(&self) -> Result<()> {
```

Comment:
Should this be removed?

Reply:
Yes! I removed list_tags
89e8c05958dc19c7827d46d8034ecd872eee3a9f